src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -5615,6 +5723,93 @@ def update_identity(self, mc: ManagedCluster) -> ManagedCluster:
             mc.identity = identity
         return mc
 
+    def update_azure_keyvault_secrets_provider_addon_profile(
+        self,
+        azure_keyvault_secrets_provider_addon_profile: ManagedClusterAddonProfile,
+    ) -> None:
+        """Update azure keyvault secrets provider addon profile in-place.
+
+        :return: None
+        """
+        # determine the value of constants
+        addon_consts = self.context.get_addon_consts()
+        CONST_SECRET_ROTATION_ENABLED = addon_consts.get(
+            "CONST_SECRET_ROTATION_ENABLED"
+        )
+        CONST_ROTATION_POLL_INTERVAL = addon_consts.get(
+            "CONST_ROTATION_POLL_INTERVAL"
+        )
+
+        if self.context.get_enable_secret_rotation():
+            azure_keyvault_secrets_provider_addon_profile.config[
+                CONST_SECRET_ROTATION_ENABLED
+            ] = "true"
+
+        if self.context.get_disable_secret_rotation():
+            azure_keyvault_secrets_provider_addon_profile.config[
+                CONST_SECRET_ROTATION_ENABLED
+            ] = "false"
+
+        if self.context.get_rotation_poll_interval() is not None:
+            azure_keyvault_secrets_provider_addon_profile.config[
+                CONST_ROTATION_POLL_INTERVAL
+            ] = self.context.get_rotation_poll_interval()
+
+    def update_addon_profiles(self, mc: ManagedCluster) -> ManagedCluster:
```

Comment:
The implementation of this function basically follows [this part](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2696-L2731) in file `custom.py`.